.goreleaser.yml

```diff
@@ -0,0 +1,71 @@
+before:
+  hooks:
+    - mkdir -p caddy-build
+    - cp cmd/caddy/main.go caddy-build/main.go
+    - cp ./go.mod caddy-build/go.mod
+    - sed -i.bkp s/github.com\/caddyserver\/caddy\/v2/caddy/g ./caddy-build/go.mod
+    # GoReleaser doesn't seem to offer {{.Tag}} at this stage, so we have to embed it into the env
+    # so we run: TAG=$(git describe --abbrev=0) goreleaser release --rm-dist --skip-publish --skip-validate
+    - go mod edit -require=github.com/caddyserver/caddy/v2@{{.Env.TAG}} ./caddy-build/go.mod
+    - git clone --depth 1 https://github.com/caddyserver/dist caddy-dist
+    - go mod download
+builds:
+- env:
+  - CGO_ENABLED=0
+  - GO111MODULE=on
+  main: main.go
+  dir: ./caddy-build
+  binary: caddy
+  goos:
+  - darwin
+  - linux
+  - windows
+  - freebsd
+  goarch:
+  - amd64
+  - 386
+  - arm
+  - arm64
+  goarm:
+  - 6
+  - 7
+  ignore:
+    - goos: darwin
+      goarch: 386
+  flags:
+  - -trimpath
+  ldflags:
+  - -s -w
+archives:
+  -
+    format_overrides:
+      - goos: windows
+        format: zip
+    replacements:
+      darwin: Mac
+      linux: Linux
+      windows: Windows
+      386: i386
+      amd64: x86_64
+checksum:
+  algorithm: sha512
```

Comment:
Any specific reason you're choosing `sha512` here? `sha256` should be enough for this purpose... I'm already using `sha256` in the [`Dockerfile`s](https://github.com/caddyserver/caddy-docker/blob/d6ce5d55b4dbff40f0452f5478fdd0d648c17a15/alpine/Dockerfile#L20), though I can certainly switch to `sha512` if need be.

Reply:
256 is probably fine, we should match the Dockerfiles.

Reply:
I researched this and [found reports](https://crypto.stackexchange.com/questions/26336/sha512-faster-than-sha256) that sha512 is faster than sha256. I was able to reproduce the performance locally. So `sha512` seemed like future-proof collision resistance for free, with rebate too.

This is my local run:
```
Doing sha256 for 3s on 16 size blocks: 11893333 sha256's in 2.93s
Doing sha256 for 3s on 64 size blocks: 6140405 sha256's in 2.90s
Doing sha256 for 3s on 256 size blocks: 2638625 sha256's in 2.93s
Doing sha256 for 3s on 1024 size blocks: 749622 sha256's in 2.86s
Doing sha256 for 3s on 8192 size blocks: 109542 sha256's in 2.96s
Doing sha512 for 3s on 16 size blocks: 9164488 sha512's in 2.91s
Doing sha512 for 3s on 64 size blocks: 9295270 sha512's in 2.96s
Doing sha512 for 3s on 256 size blocks: 3468409 sha512's in 2.97s
Doing sha512 for 3s on 1024 size blocks: 1182612 sha512's in 2.97s
Doing sha512 for 3s on 8192 size blocks: 168891 sha512's in 2.97s
LibreSSL 2.6.5
built on: date not available
options:bn(64,64) rc4(16x,int) des(idx,cisc,16,int) aes(partial) blowfish(idx)
compiler: information not available
The 'numbers' are in 1000s of bytes per second processed.
type             16 bytes     64 bytes    256 bytes   1024 bytes   8192 bytes
sha256           64946.53k   135512.39k   230541.98k   268396.13k   303164.89k
sha512           50388.94k   200978.81k   298960.51k   407742.32k   465843.46k
```

Reply:
@Mohammed90 ok, I can switch to `sha512` in the `Dockerfile`!

Reply:
Wow. Neat!